claims_p2.go

```diff
@@ -161,7 +202,16 @@ func (c P2Claims) ToCBOR() ([]byte, error) { //nolint:gocritic
 }
 
 func (c P2Claims) ToUnvalidatedCBOR() ([]byte, error) { //nolint:gocritic
+	var scs ISwComponents
+	if c.SwComponents != nil && c.SwComponents.IsEmpty() {
+		scs = c.SwComponents
+		c.SwComponents = nil
+	}
+
 	buf, err := em.Marshal(&c)
+	if scs != nil {
```

Comment:
I am not clear on the scope and rationale of setting elements inside c.SwComponents() ??

MashalCBOR Method should be ONLY responsible for Marshaling : Why make it set & re-set the elements? can you please help clarify..

Reply:
It's resetting after setting them to `nul` earlier in the method (line 208) this is to ensure the correct behavoir of `omitempty`, which, thinking about it we don't actually need here since the field isn't `omitempty`; will clean this up.